### PR TITLE
cluster: use dummy metadata when initial fetch fails

### PR DESCRIPTION
**Pull request by piodul**
Currently, the driver fetches topology and schema information (we call
it metadata) on Session creation and fails if an error occurs during
fetching/parsing of that data - despite the fact that the metadata is
not vital for the driver to work (although some information is needed
for better performance, e.g. topology info and keyspace replication
strategies).

While, in ideal world, we would like the metadata fetch to always
succeed, it can fail either due to our fault (e.g. bugs like #437) or
reasons beyond our control (connection errors, non-backwards compatible
changes in system tables). Therefore, we would like for the driver to
work even when no metadata is available.

This commit modifies the initial metadata fetch code so that, in case it
fails, a dummy metadata is used instead.

Fixes: #438

## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [ ] ~I added relevant tests for new features and bug fixes.~
- [x] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [x] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

**Comment by piodul** [comment]
I tested it by changing the `MetadataReader::read_metadata` to immediately return an error. This caused 7 tests to fail, but only those which test features which require the metadata fetch to work correctly.

The following tests failed:

```
transport::session_test::test_column_kinds_in_metadata
transport::session_test::test_prepared_partitioner
transport::session_test::test_primary_key_ordering_in_metadata
transport::session_test::test_table_partitioner_in_metadata
transport::session_test::test_user_defined_types_in_metadata
transport::session_test::test_turning_off_schema_fetching
transport::session_test::test_schema_types_in_metadata
```

If anybody has an idea on how to write an automatic test for this feature, I'll be happy to add it.

File: scylla/src/transport/cluster.rs [hunk 1]
```diff
@@ -84,28 +84,37 @@ impl Cluster {
...
+            server_events_sender,
+            fetch_schema_metadata,
+        );
+
+        let metadata = match metadata_reader.read_metadata().await {
+            Ok(data) => data,
+            Err(err) => {
+                warn!(
+                    error = ?err,
+                    "Initial metadata read failed, proceeding with dummy metadata",
```
Comment (psarna) [inline scylla/src/transport/cluster.rs:104]: let's be specific, this is a user-facing error: "proceeding with metadata consisting only of the initial peers list"

Comment (psarna) [reply scylla/src/transport/cluster.rs:104]: (or alike, feel free to rephrase)

Comment (piodul) [reply scylla/src/transport/cluster.rs]: Done, I also mentioned how the failed metadata fetch might affect the user.

**Pull request merged**